decidim-proposals/app/models/decidim/proposals/proposal.rb

```diff
@@ -179,39 +184,67 @@ def published?
         published_at.present?
       end
 
+      # Public: Returns the published state of the proposal.
+      #
+      # Returns Boolean.
+      def state
+        return amendment.state if emendation?
+        return nil unless published_state? || withdrawn?
+
+        super
+      end
+
+      alias_attribute :internal_state, :state
```

Comment:
Can you explain the why of this alias?
Isn't `internal_state` going to be overridden by the method with the same name, just below?

Reply:
Ok, I'll try to explain it. I changed `state` to expose the published state. I've also thought about keeping it as it was and creating a new `published_state`, but it would imply changing a lot of places and that could break many things.

With the modified `state` everything keeps working ok, but I need to access to the unpublished state in the admin zone, to show it in the listing and to allow to modify it. And that's where I need to create `internal_state` to be used by the [ProposalAnswerForm](https://github.com/decidim/decidim/pull/5810/files#diff-9aa03b14b04b53663ebb4cb72952f192R15). When it was using `state` it wasn't showing the internal state as selected when rendering the form. So, the setter should be the same that `state` setter, but the getter has to be different.

I tried to create an alias only for the setter, but it didn't work. Then, I've found the `alias_attribute` and I've decided to use it to generate the setter and then override the getter. 

I also could copy the `state` setter code, but I preferred the `alias_attribute` solution:
```ruby
def internal_state=(value)
  self[:state] = value
end
```

What do you think?

Reply:
It seems ok, but we need this explanation documented on the `alias_attribute` declaration, this way everybody will understand the intention when reading the code in the future. :smile: